x/group/server/invariants.go

```diff
@@ -0,0 +1,99 @@
+package server
+
+import (
+	"math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/regen-network/regen-ledger/orm"
+	"github.com/regen-network/regen-ledger/x/group"
+)
+
+func (s serverImpl) RegisterInvariants(ir sdk.InvariantRegistry) {
+	ir.RegisterRoute(group.ModuleName, "Tally-Votes", s.tallyVotesInvariant())
+}
+
+func (s serverImpl) AllInvariants() sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		return s.tallyVotesInvariant()(ctx)
+	}
+}
+
+func (s serverImpl) tallyVotesInvariant() sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		if ctx.BlockHeight()-1 < 0 {
+			return sdk.FormatInvariant(group.ModuleName, "Tally-Votes", "Not enough blocks to perform TallyVotesInvariant"), false
+		}
+		prevCtx, _ := ctx.CacheContext()
+		prevCtx = prevCtx.WithBlockHeight(ctx.BlockHeight() - 1)
+		msg, broken := tallyVotesInvariant(ctx, prevCtx, s.proposalTable)
+		return sdk.FormatInvariant(group.ModuleName, "Tally-Votes", msg), broken
+	}
+}
+
+func tallyVotesInvariant(ctx sdk.Context, prevCtx sdk.Context, proposalTable orm.AutoUInt64Table) (string, bool) {
+
+	var msg string
+	var broken bool
+
+	prevIt, err := proposalTable.PrefixScan(prevCtx, 1, math.MaxUint64)
+	if err != nil {
+		return err.Error(), true
```

Comment:
```suggestion
		return msg, broken, err
```
same for all error handling below